### PR TITLE
🐛 fixing flaky test test_checked_once_task_is_auto_removed

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Very pesky edge case
`FAILED tests/long_running_tasks/test_long_running_tasks_task.py::test_checked_once_task_is_auto_removed - Failed: DID NOT RAISE <class 'servicelib.long_running_tasks._errors.TaskNotFoundError'>`

Pytest raises a BaseException that is not caught and retried by tenacity.
Also the test logic was not 100% accurate

## Related issue/s

- closes https://github.com/ITISFoundation/osparc-simcore/issues/3671

### CODE REVIEW

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py [hunk 1]
```diff
@@ -102,9 +110,12 @@ async def test_checked_once_task_is_auto_removed(tasks_manager: TasksManager):
     )
     # check once (different branch in code)
     tasks_manager.get_task_status(task_id, with_task_context=None)
-    await asyncio.sleep(2 * TEST_CHECK_STALE_INTERVAL_S + 1)
     async for attempt in AsyncRetrying(**_RETRY_PARAMS):
         with attempt:
+            # to detect a stale task, you must not call get status,
+            # we need to sleep before retrying in order to detect it
+            await asyncio.sleep(2 * TEST_CHECK_STALE_INTERVAL_S + 1)
```
Comment (sanderegg) [inline packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py:117]: I am not sure I understand the need to add a sleep in a retry that already sleeps per se between the retries? why not just modify the wait_fixed parameter?

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py [hunk 2]
```diff
@@ -29,12 +30,17 @@
...
-    stop=stop_after_delay(60),
-    retry=retry_if_exception_type(AssertionError),
-)
+_RETRY_PARAMS = {
+    "reraise": True,
+    "wait": wait_fixed(0.1),
+    "stop": stop_after_delay(60),
+    "retry": retry_if_exception_type(
+        (
+            BaseException,  # retry when pytest reports `Failed: DID NOT RAISE``
```
Comment (sanderegg) [inline packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py:39]: really? you mean that with pytest.raises does not raise an AssertionError when there is no exception???

**Review by pcrespov — COMMENTED** [review]
Paired reviewed. Please re-assign when our discussion was revisited. Thank you.

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py (hunk 2 above)
Comment (GitHK) [reply packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py]: yeah, but this is a bad pattern, so I will avoid catching this one.

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py (hunk 1 above)
Comment (GitHK) [reply packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py]: I've rewritten the test so it should be much more readable now.
With this occasion I've also refactored and merged with the one above

**Review by sanderegg — APPROVED** [review]
thanks for fixing this!

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py [hunk 3]
```diff
@@ -75,42 +78,37 @@ async def tasks_manager() -> AsyncIterator[TasksManager]:
...
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_status(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result_old(task_id)
+            for tasks in tasks_manager._tasks_groups.values():  # noqa: SLF001
+                if task_id in tasks:
+                    msg = "wait till no element is found any longer"
+                    raise AssertionError(msg)
```
Comment (pcrespov) [inline packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py:104]: MINOR: if you want to make it more readable, you can use `raise tenacity.TryAgain(msg)`

File: packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py [hunk 4]
```diff
@@ -29,12 +30,12 @@
...
-    reraise=True,
-    wait=wait_fixed(0.1),
-    stop=stop_after_delay(60),
-    retry=retry_if_exception_type(AssertionError),
-)
+_RETRY_PARAMS: dict[str, Any] = {
+    "reraise": True,
+    "wait": wait_fixed(0.1),
+    "stop": stop_after_delay(60),
+    "retry": retry_if_exception_type(AssertionError),
```
Comment (pcrespov) [inline packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py:37]: If you use `TryAgain` it is cleaner since `assert expression` also raises `AssertionError`

**Review by pcrespov — APPROVED** [review]
Your last modifications seem more clear to me.
Some extra minor comments
